# HG changeset patch
# User Rik <rik@octave.org>
# Date 1349052338 25200
#      Sun Sep 30 17:45:38 2012 -0700
# Node ID abf34bbcd0537767bc3c00ebf199abfa5cc3bf05
# Parent  1cc42df8ede736c9af405bca29028f316eadb24f
build: Fix make distcheck on platforms that use compiled auxiliary programs.

* src/Makefile.am: Add mkoctfile.cc, octave-config.cc to list of DISTCLEANFILES.
Reorganize to have Octave binaries section before secondary compiled auxiliary
programs.

diff --git a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -29,52 +29,28 @@ AM_CPPFLAGS = \
   -I$(top_srcdir)/libinterp \
   -I$(top_builddir)/libinterp/interpfcn \
   -I$(top_builddir)/libgnu -I$(top_srcdir)/libgnu
 
 AM_CFLAGS += $(WARN_CFLAGS)
 
 AM_CXXFLAGS += $(WARN_CXXFLAGS)
 
-OCTAVE_BINARIES = \
-  octave \
-  octave-cli
-
-if AMCOND_BUILD_COMPILED_AUX_PROGRAMS
-bin_PROGRAMS = \
-  mkoctfile \
-  $(OCTAVE_BINARIES) \
-  octave-config
-
-mkoctfile_SOURCES =
-nodist_mkoctfile_SOURCES = mkoctfile.cc
-mkoctfile_LDADD = $(top_builddir)/libgnu/libgnu.la $(LIBS)
-
-octave_config_SOURCES =
-nodist_octave_config_SOURCES = octave-config.cc
-octave_config_LDADD = \
-  $(top_builddir)/libinterp/corefcn/libcorefcn.la \
-  $(top_builddir)/libgnu/libgnu.la \
-  $(LIBS)
-
-else
-bin_PROGRAMS = \
-  $(OCTAVE_BINARIES)
-
-bin_SCRIPTS = \
-  mkoctfile \
-  octave-config
-endif
-
 EXTRA_DIST = \
   mkoctfile.in.cc \
   mkoctfile.in.sh \
   octave-config.in.cc \
   octave-config.in.sh
 
+DISTCLEANFILES =
+
+OCTAVE_BINARIES = \
+  octave \
+  octave-cli
+
 OCTAVE_CORE_LIBS = \
   $(top_builddir)/libinterp/liboctinterp.la \
   $(top_builddir)/liboctave/liboctave.la
 
 include ../libgui/link-deps.mk
 
 octave_cli_SOURCES = main-cli.cc
 
@@ -105,21 +81,50 @@ octave_LDADD = \
   $(OCTAVE_CORE_LIBS) \
   $(OCTAVE_GUI_LINK_DEPS)
 
 octave_LDFLAGS = \
   $(NO_UNDEFINED_LDFLAG) \
   $(OCTAVE_GUI_LINK_OPTS)
 
 if AMCOND_BUILD_COMPILED_AUX_PROGRAMS
+bin_PROGRAMS = \
+  $(OCTAVE_BINARIES) \
+  mkoctfile \
+  octave-config
+
+mkoctfile_SOURCES =
+nodist_mkoctfile_SOURCES = mkoctfile.cc
+mkoctfile_LDADD = $(top_builddir)/libgnu/libgnu.la $(LIBS)
+
+octave_config_SOURCES =
+nodist_octave_config_SOURCES = octave-config.cc
+octave_config_LDADD = \
+  $(top_builddir)/libinterp/corefcn/libcorefcn.la \
+  $(top_builddir)/libgnu/libgnu.la \
+  $(LIBS)
+
+DISTCLEANFILES += mkoctfile.cc octave-config.cc
+
+else
+bin_PROGRAMS = \
+  $(OCTAVE_BINARIES)
+
+bin_SCRIPTS = \
+  mkoctfile \
+  octave-config
+endif
+
+if AMCOND_BUILD_COMPILED_AUX_PROGRAMS
 octave-config.cc: octave-config.in.cc Makefile
 	@$(do_subst_default_vals)
 
 mkoctfile.cc: mkoctfile.in.cc Makefile
 	@$(do_subst_config_vals)
+
 else
 octave-config: octave-config.in.sh Makefile
 	@$(do_subst_default_vals)
 	chmod a+rx $@
 
 mkoctfile: mkoctfile.in.sh Makefile
 	@$(do_subst_config_vals)
 	chmod a+rx $@
